query/src/org/labkey/query/QueryServiceImpl.java

```diff
@@ -264,6 +271,125 @@ public WhereClause createFilterClause(@NotNull FieldKey fieldKey, Object value)
         }
     };
 
+    private static SQLFragment getColumnInSql(@NotNull FieldKey fieldKey, Set<String> params, User user, Container container, Map<FieldKey, ? extends ColumnInfo> columnMap, boolean negate)
+    {
+        ColumnInfo col = columnMap.get(fieldKey);
+        SQLFragment colFrag = new SQLFragment(col == null ? fieldKey.toString() : col.getAlias());
+        String[] values = params.toArray(new String[0]);
+
+        if (values.length < 3)
+            throw new NotFoundException("Incorrect number of arguments provided");
```

Comment:
nit: `NotFoundException` seems to apply more to a resource reference (e.g., 404 error) vs a missing/malformed parameter or argument

```suggestion
            throw new IllegalArgumentException("Incorrect number of arguments provided");
```